OsmAnd/src/net/osmand/plus/routepreparationmenu/cards/RouteDirectionsCard.java

```diff
@@ -43,6 +45,7 @@ protected void updateContent() {
 
 	private void createRouteDirections(LinearLayout cardsContainer) {
 		List<RouteDirectionInfo> routeDirections = app.getRoutingHelper().getRouteDirections();
+		removeUnnecessaryGoAhead(routeDirections);
```

Comment:
Place to fix is very awkward of course. Also the bug is about our routing i.e. Attach to roads, so we should not produce unnecessary go ahead in the first place!